WearComplicationProvidersTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/LongTextProviderService.kt

```diff
@@ -53,52 +59,138 @@ class LongTextProviderService : SuspendingComplicationProviderService() {
-                        // should always be displayed by the watch face. This means that when a long text
-                        // provider supplies both title and icon, it is expected that both are displayed.
-                        setLongText(ComplicationText.plainText(getString(R.string.long_text_with_icon_and_title)))
-                        setLongTitle(ComplicationText.plainText(getString(R.string.long_title)))
-                        setIcon(
-                            Icon.createWithResource(
-                                this@LongTextProviderService,
-                                R.drawable.ic_battery
-                            )
-                        )
-                        setBurnInProtectionIcon(
-                            Icon.createWithResource(
-                                this@LongTextProviderService,
-                                R.drawable.ic_battery_burn_protect
-                            )
-                        )
-                    }
-                    3 -> {
-                        setLongText(ComplicationText.plainText(getString(R.string.long_text_with_title)))
-                        setLongTitle(ComplicationText.plainText(getString(R.string.long_title)))
-                    }
-                    4 -> {
-                        setLongText(ComplicationText.plainText(getString(R.string.long_text_with_image)))
-                        setSmallImage(Icon.createWithResource(this@LongTextProviderService, R.drawable.outdoors))
-                    }
-                    5 -> {
-                        setLongText(ComplicationText.plainText(getString(R.string.long_text_with_image_and_title)))
-                        setLongTitle(ComplicationText.plainText(getString(R.string.long_title)))
-                        setSmallImage(Icon.createWithResource(this@LongTextProviderService, R.drawable.aquarium))
-                    }
-                }
+        val case = Case.values()[state.mod(Case.values().size)]
+        return getComplicationData(
+            tapAction = complicationTogglePendingIntent,
+            case = case
+        )
+    }
+
+    override fun getPreviewData(type: ComplicationType): ComplicationData =
+        getComplicationData(
+            tapAction = null,
+            case = Case.TEXT_WITH_ICON_AND_TITLE
+        )
+
+    private fun getComplicationData(
+        tapAction: PendingIntent?,
+        case: Case
+    ): ComplicationData {
+        val text: ComplicationText
+        val smallImage: SmallImage?
+        val title: ComplicationText?
+        val monochromaticImage: MonochromaticImage?
+        val contentDescription: ComplicationText
+
+        when (case) {
+            Case.TEXT_ONLY -> {
+                text = PlainComplicationText.Builder(
+                    text = getText(R.string.long_text_only)
+                ).build()
+                smallImage = null
```

Comment:
Kotlin tends to embrace terseness. In that spirit I'd suggest doing something like this:

private fun getComplicationData(tapAction: PendingIntent?, case: Case) = when (case) {
  Case.TEXT_ONLY -> LongTextComplicationData.Builder(
            text = PlainComplicationText.Builder(
                    text = getText(R.string.long_text_only)
                ).build(),
            contentDescription = PlainComplicationText.Builder(
                    text = getText(R.string.long_text_only_content_description)
                ).build()
        ).build()

  Case.TEXT_WITH_ICON -> etc...
}

Reply:
Updated most of the builders to follow this pattern in 629b4e8!